### PR TITLE
Add aptly execution module

### PR DESCRIPTION
**Pull request by m03**
### What does this PR do?

* Adds an execution module with minimal functionality for aptly Debian repository manager.

### What issues does this PR fix or reference?

* None

### Tests written?

No

### Notes

* Tested against aptly 0.96 (the version available in Ubuntu 16.04).
* Since I'm still actively working on both the state and execution modules, and the unit tests, some of the function names and implemented functionality will likely need minor refactoring in the near future.

### CODE REVIEW

File: salt/modules/aptly.py [hunk 1]
```diff
@@ -0,0 +1,517 @@
...
+def __virtual__():
+    '''
+    Only works on systems with the aptly binary in the system path.
+    '''
+    if salt.utils.path.which('aptly'):
+        return __virtualname__
+    return (False, 'The aptly binaries required cannot be found or are not installed.')
+
+
+def _cast_if_numeric(value):
```
Comment (cachedout) [inline salt/modules/aptly.py]: We could likely use this as a general utility function. Perhaps we should move this into the `salt.utils` family?

File: salt/modules/aptly.py [hunk 2]
```diff
@@ -0,0 +1,517 @@
...
+
+    _LOG.debug('Found repositories: %s', len(repos))
+
+    for name in repos:
+        ret[name] = get_repo(name=name, config_path=config_path,
+                             with_packages=with_packages)
+    return ret
+
+
+def get_repo(name, config_path=_DEFAULT_CONFIG_PATH, with_packages=False):
```
Comment (cachedout) [inline salt/modules/aptly.py]: Maybe `return_packages` would be a better name? Not sure.

**Review by cachedout — COMMENTED** [review]
Very nice! I just left some really minor feedback here. Thanks!

File: salt/modules/aptly.py (hunk 2 above)
Comment (m03) [reply salt/modules/aptly.py]: I was trying to stay as close to the aptly command line argument names as possible. Plus, `return_packages` might give the impression that it returns only the packages in the repo and not other metadata.

File: salt/modules/aptly.py (hunk 1 above)
Comment (m03) [reply salt/modules/aptly.py]: @cachedout I just checked and `stringutils` does have something similar (`to_num`), though it doesn't handle empty strings the way I would like. I'll add a commit to add the functionality I need to `stringutils`.

File: salt/modules/aptly.py (hunk 2 above)
Comment (cachedout) [reply salt/modules/aptly.py]: Sounds good. Thanks!

**Review by cachedout — APPROVED** [review]

**Comment by m03** [comment]
It's not clear why the `salt-pr-linode-ubuntu16-py3` step failed, as the test results in Jenkins appear to state that there were no failures.

**Comment by rallytime** [comment]
Tests are all green now. Looks like we just had a problem getting the test report back. Thanks for submitting this @m03.

**Pull request merged**